### PR TITLE
Add rest API to query for providers

### LINKED ISSUE #12468: Add API to query for providers

### PR DESCRIPTION
**Pull request by nic314**
closes: #12468 

Added a new api rest endpoint in the stable api to query the installed providers. It provides the same information as the cli command "airflow providers list"

Note:  There is some code duplication between provider_endpoint.py and  provider_command.py regarding_remove_rst_syntax and the mapping, i kept it like this becouse i assume that in future its possible that the rest api and the cli returns different informations/fields but if that's not the case and the rest and cli will always return the same information then i can refactor my pr to avoid this duplication.

### CODE REVIEW

**Comment by nic314** [comment]
i see that "build docs" and "Sqlite Py3.6: Always Core Other API CLI Providers WWW Integration Heisentests" failed, i tried locally to run those steps on the master branch and i have the same exact error, seems like they are not related to this PR. 
My first PR though so i may be wrong :)

File: airflow/api_connexion/openapi/v1.yaml
Comment (mik-laj) [inline airflow/api_connexion/openapi/v1.yaml:853]: Can you add this change to "Summary of changes"? http://apache-airflow-docs.s3-website.eu-central-1.amazonaws.com/docs/apache-airflow/latest/stable-rest-api-ref.html#section/Summary-of-Changes

**Comment by potiuk** [comment]
Docs also need to be fixed.

File: airflow/api_connexion/openapi/v1.yaml
Comment (nic314) [reply airflow/api_connexion/openapi/v1.yaml:853]: I added a new line in that summary table, let me know if its fine .

**Comment by nic314** [comment]
@potiuk You mean the step in the CI that was failing? Or i missed to update some documentation files?
I rebased from master, now the only failure is in TestAwsBatchOperator but locally im able to reproduce the same failure on the master branch so i don't think that it's connected to this PR

**Comment by potiuk** [comment]
Yep. We just merged a fix for that AWS batch. If f you could rebase the last time (I hope) that would be great!

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
I think we should refrain from merging it until we agree on how we approach 2.0.1 with branching: https://github.com/apache/airflow/pull/13382#issuecomment-752570884

**Review by potiuk — CHANGES_REQUESTED** [review]
Looks good, but seems this is the first change to be merged that should be targeted to 2.1  so we need to decide how we approach it.

**Comment by jhtimmins** [comment]
@kaxil @ephraimbuddy This looks good to me, but can y'all confirm that this is still good to go? If so, we can merge it tomorrow.

@nic314 Sorry this has been left in limbo for so long. Would you mind rebasing it onto master?

**Comment by nic314** [comment]
@jhtimmins   I rebased, but the cicd needs some approval.  Also i was assuming this was for the version 2.1, let me know is i need to change here https://github.com/apache/airflow/pull/13394/commits/c4c5a6f89df01b70bbc09535e6b7326107e6285e#diff-191056f40fba6bf5886956aa281e0e0d2bb4ddaa380beb012d922a25f5c65750R136

File: tests/api_connexion/endpoints/test_provider_endpoint.py
Comment (ephraimbuddy) [inline tests/api_connexion/endpoints/test_provider_endpoint.py:66]: A while ago, we made some changes to tests speedup. I'd appreciate a change of this test to use that. See https://github.com/apache/airflow/blob/b7b17641d0c17bfa5e1f8c77b9c232da73bbedcd/tests/api_connexion/endpoints/test_connection_endpoint.py

**Review by ephraimbuddy — CHANGES_REQUESTED** [review]
Just the tests, LGTM overrall

File: tests/api_connexion/endpoints/test_provider_endpoint.py
Comment (nic314) [reply tests/api_connexion/endpoints/test_provider_endpoint.py:66]: i will do asap, i think within today

**Comment by ephraimbuddy** [comment]
> Looks good, but seems this is the first change to be merged that should be targeted to 2.1 so we need to decide how we approach it.

Hi @potiuk , is it time we merge this?

**Comment by jhtimmins** [comment]
@ephraimbuddy We can go ahead and merge, since this has targeted 2.1 and  we're actively trying to get those PRs in.

**Comment by potiuk** [comment]
> @ephraimbuddy We can go ahead and merge, since this has targeted 2.1 and we're actively trying to get those PRs in.

I just reverted the offending PR (and asked the author to fix also the reason why (https://github.com/apache/airflow/pull/15703#issuecomment-833894099)  

hey @jhtimmins - broken master happens from time to time for different reasons, so far we've been trying to fix such things (often by just pin-pointing and reverting such offending PRs'). I think we should keep doing it, Merging "quickly" something because we are in hurry is almost never a good idea.

**Comment by potiuk** [comment]
I think so far in Airflow we were doing really well about avoiding the "not my problem" attitude.

**Comment by jhtimmins** [comment]
@potiuk Thanks for the heads up about master, I missed that it was failing. 

As I understood it, Ephraim was asking whether we _actually_ wanted this to be included for 2.1. As I'm in the middle of working on the 2.1 release, and am currently making a list of what will get into the release versus pushed to a later one, I was confirming that this feature should be included. Since the release date is approaching, I was encouraging him to be proactive about getting it fully reviewed and merged, not to cut any corners in quality or good practices.

**Comment by potiuk** [comment]
> @potiuk Thanks for the heads up about master, I missed that it was failing.
> 
> As I understood it, Ephraim was asking whether we _actually_ wanted this to be included for 2.1. As I'm in the middle of working on the 2.1 release, and am currently making a list of what will get into the release versus pushed to a later one, I was confirming that this feature should be included. Since the release date is approaching, I was encouraging him to be proactive about getting it fully reviewed and merged, not to cut any corners in quality or good practices.

Sorry then :). I misunderstood that - It looked as if we are hurrying now for unknown reason :).

**Comment by potiuk** [comment]
Hey @nic314 - can you please rebase. I am also happy to merge it as soon as it is green!

**Review by potiuk — APPROVED** [review]
Looks great! Thanks @nic314 . Let's hope all tests pass this time :).

**Review by ephraimbuddy — APPROVED** [review]

**Comment by potiuk** [comment]
Wohoo :tada: !

**Pull request merged**
